pkg/api/controller.go

```diff
@@ -1238,19 +1242,28 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 	writeResponse(w, http.StatusCreated, response)
 }
 
-func ensureStorageNamespace(ctx context.Context, adapter block.Adapter, storageNamespace string) error {
+var errorStorageNamespaceInUse = errors.New("lakeFS repositories can't share storage namespace")
+
+func (c *Controller) ensureStorageNamespace(ctx context.Context, storageNamespace string) error {
 	const (
 		dummyKey  = "dummy"
 		dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
 	)
 
 	obj := block.ObjectPointer{StorageNamespace: storageNamespace, Identifier: dummyKey}
 	objLen := int64(len(dummyData))
-	err := adapter.Put(ctx, obj, objLen, strings.NewReader(dummyData), block.PutOpts{})
-	if err != nil {
+	if _, err := c.BlockAdapter.Get(ctx, obj, objLen); err == nil {
+		return fmt.Errorf("found lakeFS objects in the storage namespace(%s): %w",
```

Comment:
```suggestion
		return fmt.Errorf("%s: %w",
```
(and maybe rephrase the error itself if that doesn't look nice enough).

Reply:
Would rather keep it as is. The first sentence makes it clearer IMO.